### PR TITLE
[RPA-1545] Assigning user to tenant - frontend & fixes

### PR DESCRIPTION
**Pull request by Aeri4a**
## Description

Tenant Admin view on users
image
image

Admi view on users
image
image

Tenants invite link
image
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java [hunk 1]
```diff
@@ -115,6 +115,7 @@ public TenantDTO save(TenantDTO tenantDTO) {
 
         Tenant tenant = tenantMapper.toEntity(tenantDTO);
         tenant.setCreatedBy(currentUser);
+        tenant.setLastModifiedBy("system");
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java]: why don't we update it with the user that saves this tenant?

Comment (star-in-the-jar) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java]: also why last_modified_by is a string type, not IUser?

Comment (star-in-the-jar) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java]: or userId: Int at least?

File: runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java [hunk 2]
```diff
@@ -164,13 +165,46 @@ public void delete(UUID id) {
...
+                .findByTenantIdAndExpirationDateBefore(
+                    tenantId, ZonedDateTime.now()
+                ).map(TenantInviteCodeDTO::new);
+        }
+
+        return tenantInviteCodeRepository
+            .findByTenantIdAndExpirationDateBefore(
+                user.getTenant().getId(), ZonedDateTime.now()
+            ).map(TenantInviteCodeDTO::new);
+    }
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java:185]: keep this, but in the future let's breake cases like that (when users with admin/user role can ask for different stuff) divide it into two endpoints depending on who is asking (by role)

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/list/edit.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ const editListTranslations = {
     'Users.List.Edit.Form.Select.Language.English': 'English',
     'Users.List.Edit.Form.Select.Language.Polish': 'Polish',
     'Users.List.Edit.Form.Switch.Activated': 'Activated:',
+    'Users.List.Edit.Form.Switch.InfoTooltip': 'To change the role deactivate the user',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/list/edit.ts:17]: 👍

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/register.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
     "Register.Form.Fields.Email.Label": "Email",
     "Register.Form.Fields.Password.Label": "Hasło",
     "Register.Form.Fields.PasswordConfirmation.Label": "Powtórz hasło",
+    "Register.Form.Fields.InviteCode.Label": "Kod zaproszenia",
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/register.json:18]: so "link" or "kod"?  take a look at components.json

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersBrowseView/UsersBrowseViewHeader.tsx [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
 import { StyledHeaderGrid } from './UsersBrowseView.styles';
 
 const UsersBrowseViewHeader: FC = () => {
     const { translate } = useTranslations();
+    const { user }= useAuth();
+    const hasAdminAccess = useRole([Role.ROLE_ADMIN]);
+
+    const header = hasAdminAccess
+        ? translate('Users.Browse.Header.Title')
+        : `${translate('Users.Browse.Header.Title')} - ${user.tenant.name}`;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersBrowseView/UsersBrowseViewHeader.tsx]: imo | looks better than -

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListEdit/UsersListEdit.types.ts [hunk 6]
```diff
@@ -4,6 +4,7 @@ export interface UsersListEditDialogProps {
     open: boolean;
     onClose: () => void;
     userData: IUser;
+    isForAdmin: boolean;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListEdit/UsersListEdit.types.ts:7]: does it mean that it's visible for admins only?

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListEdit/UsersListEdit.types.ts:7]: nvm

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx [hunk 7]
```diff
@@ -63,24 +93,30 @@ const UsersRegistrationView: FC = () => {
...
     };
 
-    const mapUserActivateRequest = (id: number, role: Role): MapActivatedUserParams => {
-        const { email } = notActivated.allByPage.content.find((row) => row.id === id);
-        return { id, email, roles: [role], activated: true };
+    const mapUserActivateRequest = (id: number, role: Role, tenantId: string): MapActivatedUserParams => {
+        const { email } = hasAdminAccess
+            ? notActivated.allByPage.content.find((row) => row.id === id)
+            : tenantNotActivated.allByPage.content.find((row) => row.id === id);
+        return { id, email, roles: [role], activated: true, ...(tenantId && { tenant: { id: tenantId } }) };
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx]: add newlines after each ','

**Review by star-in-the-jar — APPROVED** [review]
+version if not bumped

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/register.json (hunk 4 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/register.json:18]: In InviteCodeButton component we are generating link. But if we travel to register section by this link then code from link in url is pasted into the field.

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersBrowseView/UsersBrowseViewHeader.tsx (hunk 5 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersBrowseView/UsersBrowseViewHeader.tsx]: changed

File: runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java (hunk 1 above)
Comment (Aeri4a) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java]: I'm not sure if we need a relation here. For now, I added here requester email.

**Review by star-in-the-jar — APPROVED** [review]
version bump

**Pull request merged**